app/models/concerns/rbac.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Role-based Access Control
+module Rbac
+  extend ActiveSupport::Concern
+
+  attr_accessor :loaded_permissions
+
+  def permission
+    return loaded_permissions if loaded_permissions.present?
+
+    self.loaded_permissions = parse_rbac_permissions(request_permission_data)
+  end
```

Comment:
Below you're using memoization, but here you're juggling with attr_accessor. Why?

Reply:
Also why singular? Is it retrieving a single permission?